### ISSUE TITLE
libicuuc.so.66: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by gitting-around** [INTO-CPS-Association/fmu-rabbitmq#56]
libicuuc.so.66: cannot open shared object file: No such file or directory for ubuntu 21. Jakob provided a fix [see here](https://gitlab.au.dk/desktop_robotti/drobotti-intocps/-/blob/84eef5a1afbf72947c82b65a44ff915a9e40f3ba/README.md)

libicuuc.so.70: cannot open shared object file: No such file or directory for ubuntu 20. We tried to fix by installing that version instead, [from here](http://archive.ubuntu.com/ubuntu/pool/main/i/icu/), and go the following error:

```bash
:~/Downloads$ sudo dpkg -i libicu70_70.1-2_amd64.deb 
Selecting previously unselected package libicu70:amd64.
(Reading database ... 272272 files and directories currently installed.)
Preparing to unpack libicu70_70.1-2_amd64.deb ...
Unpacking libicu70:amd64 (70.1-2) ...
dpkg: dependency problems prevent configuration of libicu70:amd64:
 libicu70:amd64 depends on libc6 (>= 2.34); however:
  Version of libc6:amd64 on system is 2.31-0ubuntu9.9.
 
dpkg: error processing package libicu70:amd64 (--install):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
Errors were encountered while processing:
 libicu70:amd64
```

**Comment by hejersbo** [comment]
The latest rmqfmu release (v.2.1.5) is built with a dependency on libicuuc.so.66 which would match the version present on Ubuntu 20.04 (LTS). So for the issue regarding Ubuntu 20 I would recommend trying the latest v.2.1.5 release of the rmqfmu and NOT update to later versions of libicuuc.so since that will not work. I am assuming the reported problem is caused by a locally/latest build of the rmqfmu (not released) and applied on Ubuntu 20.

**Comment by clagms** [comment]
I understand it's not possible to include all dependencies, but maybe we can take this as evidence that this one should be shipped with the rabbitmqfmu? Apologies if this doesn't make sense...

**Comment by gitting-around** [comment]
Another thing, I don't get this error on my ubuntu 20 vm. So it seems a bit random which installation has the library which hasn't. But I agree with you Henrik, we need to figure which has the dependency and remove if not used.

**Comment by gitting-around** [comment]
spoke too soon :D, I did get it, which is good, at least we don't get errors willy nilly.

**Comment by hejersbo** [comment]
We agree that this external dependency should either be removed (if not needed) or included in the fmu (if we can statically link it or provide it as thirdparty). But I would still like to figure why the Ubuntu 20 issue occurs with the latest rmqfmu release (v.2.1.5). Gita since you can replicate this, can you please provide the output of the following two commands on your Ubuntu 20 system: 

1. `ldd binaries/linux64/rabbitmq.so` (in the unzipped rmqfmu2.1.5 release)
2. `ls -al /lib/x86_64-linux-gnu/libicu*`

**Comment by gitting-around** [comment]
For the first one:

```bash
(required by fmus/rabbit-log/binaries/linux64/rabbitmq.so)
	linux-vdso.so.1 (0x00007fff6c1de000)
	libicuuc.so.70 => not found
	libicudata.so.70 => not found
	libssl.so.3 => not found
	libcrypto.so.3 => not found
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc15a3fa000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc15a2a9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc15a28e000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc15a09c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc15aaa4000)
```

For the second one:
```bash
-rw-r--r-- 1 root root 28036420 Nov 23  2021 /lib/x86_64-linux-gnu/libicudata.a
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicudata.so -> libicudata.so.66.1
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicudata.so.66 -> libicudata.so.66.1
-rw-r--r-- 1 root root 28046896 Nov 23  2021 /lib/x86_64-linux-gnu/libicudata.so.66.1
-rw-r--r-- 1 root root  7404524 Nov 23  2021 /lib/x86_64-linux-gnu/libicui18n.a
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicui18n.so -> libicui18n.so.66.1
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicui18n.so.66 -> libicui18n.so.66.1
-rw-r--r-- 1 root root  3132040 Nov 23  2021 /lib/x86_64-linux-gnu/libicui18n.so.66.1
-rw-r--r-- 1 root root    90182 Nov 23  2021 /lib/x86_64-linux-gnu/libicuio.a
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicuio.so -> libicuio.so.66.1
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicuio.so.66 -> libicuio.so.66.1
-rw-r--r-- 1 root root    59704 Nov 23  2021 /lib/x86_64-linux-gnu/libicuio.so.66.1
-rw-r--r-- 1 root root   134598 Nov 23  2021 /lib/x86_64-linux-gnu/libicutest.a
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicutest.so -> libicutest.so.66.1
lrwxrwxrwx 1 root root       18 Nov 23  2021 /lib/x86_64-linux-gnu/libicutest.so.66 -> libicutest.so.66.1
-rw-r--r-- 1 root root    81552 Nov 23  2021 /lib/x86_64-linux-gnu/libicutest.so.66.1
-rw-r--r-- 1 root root   407686 Nov 23  2021 /lib/x86_64-linux-gnu/libicutu.a
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicutu.so -> libicutu.so.66.1
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicutu.so.66 -> libicutu.so.66.1
-rw-r--r-- 1 root root   220232 Nov 23  2021 /lib/x86_64-linux-gnu/libicutu.so.66.1
-rw-r--r-- 1 root root  3789878 Nov 23  2021 /lib/x86_64-linux-gnu/libicuuc.a
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicuuc.so -> libicuuc.so.66.1
lrwxrwxrwx 1 root root       16 Nov 23  2021 /lib/x86_64-linux-gnu/libicuuc.so.66 -> libicuuc.so.66.1
-rw-r--r-- 1 root root  1976648 Nov 23  2021 /lib/x86_64-linux-gnu/libicuuc.so.66.1
```

Wouldn't this suggest that the latest one is built with a dependency on version 70 for libicu, whereas 66 is required for ubuntu20 (and 21 also I think)?

**Comment by gitting-around** [comment]
I will test that now, thanks Henrik!

**Comment by hejersbo** [comment]
When configuring cmake on linux you should see this for xerces (otherwise I updated the workflow yml incorrectly :-) ):
```
-- Xerces-C++ configuration summary
-- --------------------------------
--
--   Version:                   3.2.2
--   Library interface version: 3.2
--
--   Installation directory:    /home/hejersbo/fmu-rabbitmq/build/external/linux-x86_64
--   C compiler:                /usr/bin/cc
--   C++ compiler:              /usr/bin/c++
--
--   Build shared libraries:    OFF
--   Path delimiters:           "/"
--   File Manager:              POSIX
--   Mutex Manager:             nothreads
--   Transcoder:                gnuiconv
--   NetAccessor:               OFF
--   Message Loader:            inmemory
--   XMLCh type:                char16_t
-- Configuring done
-- Generating done
-- Build files have been written to: /home/hejersbo/fmu-rabbitmq/build/linux-x64/xerces-c

```

**Comment by hejersbo** [comment]
These should now be the only dependencies:
```
hejersbo@D42857:~/fmu-rabbitmq$ ldd build/linux-x64/rabbitmq-fmu/rabbitmq.so
        linux-vdso.so.1 (0x00007fff119f2000)
        libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f90b3f0e000)
        libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f90b3acc000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f90b38a2000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f90b37bb000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f90b379b000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f90b3571000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f90b4489000)
```

**Comment by hejersbo** [comment]
I also made a change moving to use ssl as static lib instead of shared. So now the deps are:
```
hejersbo@D42857:~/fmu-rabbitmq$ ldd build/linux-x64/rabbitmq-fmu/rabbitmq.so
        linux-vdso.so.1 (0x00007ffc3bb8d000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007feddd914000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007feddd82d000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007feddd80d000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007feddd5e5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fedde4dc000)

```

**Comment by hejersbo** [comment]
Fixed by commit 187211e
